packages/firestore/src/api/timestamp.ts

```diff
@@ -92,4 +94,25 @@ export class Timestamp {
       ')'
     );
   }
+
+  /**
+   * Converts this object to a primitive string and returns it.
+   *
+   * Overriding this method allows Timestamp objects to be compared in JavaScript using the
+   * arithmetic comparison operators, such as `<` and `>`.
+   *
+   * See https://github.com/firebase/firebase-js-sdk/issues/2632.
+   */
+  valueOf(): string {
+    // This method returns a string of the form <seconds>.<nanoseconds> where <seconds> is
+    // translated to have a non-negative value and both <seconds> and <nanoseconds> are left-padded
+    // with zeroes to be a consistent length. Strings with this format then have a lexiographical
+    // ordering that matches the expected ordering. The <seconds> translation is done to avoid
+    // having a leading negative sign (i.e. a leading '-' character) in its string representation,
+    // which would affect its lexiographical ordering.
+    const adjustedSeconds = this.seconds - MIN_SECONDS;
+    const formattedSeconds = String(adjustedSeconds).padStart(12, '0');
```

Comment:
Maybe add comment that the year 10,000 as 12 digits.

Reply:
Done